docs/design/unsupported-args-design.md

```diff
@@ -0,0 +1,132 @@
+# Unsupported Arguments schema and common design
+Date: 2024-05-29
+
+## Abstract
+OADP allows to configure it's components parameters via number of DataProtectionApplication Schema.
+
+Unsupported arguments within the OADP operator are those that do not conform to the standard DataProtectionApplication Schema or are explicitly identified as unsupported or custom arguments.
+
+There is need to provide a common way for defining and managing unsupported arguments within the OADP operator for its' components such as velero server or node-agent arguments.
+
+## Background
+The OADP allows configuration of several predefined arguments via the DataProtectionApplication Schema. These parameters are carefully selected and tested to meet the highest quality standards. However, in some circumstances, there is a need to pass other parameters that may be under development, not yet fully tested, or specifically overriding the default ones.
+
+Over the past OADP releases, some implementations have provided access to extra arguments. However, there has been no standardization, and unknown to DataProtectionApplication Schema arguments were not allowed.
+
+## Goals
+- Define common way to allow passing extra arguments to various executables used by OADP.
+- Ensure the new common way do not interfere with already existing implementations.
+
+## Non Goals
+- Deprecate current ways of defining extra arguments.
+- Change precedences within current ways of defining extra arguments.
+- Validation of unsupported args, they are passed to the appropriate executables without any prior validation.
+
+## Design
+Unsupported arguments are defined as a `ConfigMap` within the same Namespace as OADP Operator.
+
+Multiple `ConfigMaps` may exist to represent specific sections of the OADP configuration, such as dedicated `ConfigMap` instances for Velero server arguments or node-agent arguments.
+
+The user must provide `DPA` annotation(s) with the `name` of the `ConfigMap` where the OADP configuration for the unsupported arguments is present.
+
+When ConfigMap is found, controller takes those values as the highest priority and replaces **all** of the other configuration options, even if they were defined within `DPA` schema for the section for which the `ConfigMap` was created.
+
+The Schema of an `ConfigMap` includes data that corresponds to the arguments passed to the relevant executable as in the below example:
+
+<a name="notes"></a>
+> ℹ️ **Note 1:** If an argument name is passed without `-` prefix or `--` prefix, then `--` is added to the argument name.
+
+> ℹ️ **Note 2:** If an argument value is passed without sourrounding `'`, then that single quote is added to the argument value.
```

Comment:
also check for surrounding `"`?

Reply:
That is an option, but then we need additional logic to:
1. Do the same as single quotes `'` so check if the `"` is at the beginning `and` end of the string
2. If there is we do nothing (not even putting single quote, because that would cause problems).

Reply:
Done.